src/date.rs

```diff
@@ -206,13 +206,10 @@ impl Date {
     /// assert_eq!(d.to_string(), "2022-06-07");
     /// ```
     pub fn from_timestamp(timestamp: i64, require_exact: bool) -> Result<Self, ParseError> {
-        let (timestamp_second, _) = Self::timestamp_watershed(timestamp)?;
-        let d = Self::from_timestamp_calc(timestamp_second)?;
-        if require_exact {
-            let time_second = timestamp_second.rem_euclid(86_400);
-            if time_second != 0 {
-                return Err(ParseError::DateNotExact);
-            }
+        let (timestamp_second, micros) = Self::timestamp_watershed(timestamp)?;
```

Comment:
Just a small nit, can we either do `(second, microsecond)` or `(timestamp_second, timestamp_microsecond)`?